### ISSUE TITLE
Provide content negotiation on individual resources 

### ISSUE THREAD

**Issue by jschnasse** [hbz/lobid-resources#454]
At the moment [lobid-api v2](https://lobid.org/resources) only provides JSON-LD and HTML for individual resources.

    curl http://lobid.org/resources/HT018472857 -H"accept:text/html"
    curl http://lobid.org/resources/HT018472857 -H"accept:application/json"

It would be nice to have provided more serializations like `rdf+xml`, `turtle`, `ntriples` ...  You could also consider to [support links for the different formats](https://www.mnot.net/blog/2012/10/29/NO_OPTIONS) and make them discoverable as link headers via `HTTP HEAD`.

**Comment by fsteeg** [comment]
For consistency, this will require the same functionality for lobid-organisations.

**Comment by fsteeg** [comment]
Deployed to stage:

```
curl --header "Accept: application/rdf+xml" http://stage.lobid.org/resources/HT018472857
curl --header "Accept: text/turtle" http://stage.lobid.org/resources/HT018472857
curl --header "Accept: application/n-triples" http://stage.lobid.org/resources/HT018472857
```

http://stage.lobid.org/resources/HT018472857.rdf
http://stage.lobid.org/resources/HT018472857.ttl
http://stage.lobid.org/resources/HT018472857.nt

After another talk with @jschnasse I went back to the original plan to only support this for single resources, not for `/search` requests (unlike what I mentioned in our call today).

Documentation at http://stage.lobid.org/resources/api#jsonld

**Comment by acka47** [comment]
Just took a quick look, will review thoroughly tomorrow. However, we should add namespaces to make the turtle and probably also the RDF/XML better readable.

**Comment by fsteeg** [comment]
I've set up Jena to use its extended built-in namespace prefixes and redeployed:

http://stage.lobid.org/resources/HT018472857.ttl

I don't think we should add additional, custom prefixes for this, since we'd have to maintain them in parallel to our context. For the RDF serializations we should be careful not to add additional maintenance overhead. We should restrict ourselves to what we get for free using the RDF tools.

This leads to the RDF/XML namespaces you mentioned. I guess you mean the attribute values (the elements are all in the `rdf` namespace)? Jena does not change anything about these with the built-in prefixes I enabled. In general, I don't think we should worry about (human) readability of RDF/XML.

**Comment by jschnasse** [comment]
Given that your JSON is very human readable (main purpose of turtle) I'd like to support the last comment of @fsteeg , btw. I noticed that due to the usage of mime type `application/ld+json` the json output is not longer displayed directly in my browser (I use firefox with JsonView plugin), instead a download dialog pops up. Display in browser is often a useful feature for outsiders to become familiar with the data... also important for communication in general. 
So I think it would be desirable to have at least one format that is can be displayed in browser. I would prefer JSON.

**Comment by jschnasse** [comment]
Just one fast thought:
>  don't think we should add additional, custom prefixes for this
Could it be an option to take namespaces from the context?

**Comment by acka47** [comment]
I think we should add custom namespaces and don't see much maintenance effort for this. Namespaces are on the vocabulary level while the context has to be adjusted with each change for a property or class we use.

I suggest removing the built-in prefixes and use this list instead:

```
@prefix bf: <http://id.loc.gov/ontologies/bibframe/> .
@prefix bibo: <http://purl.org/ontology/bibo/> .
@prefix dc:    <http://purl.org/dc/elements/1.1/> .
@prefix dcterms: <http://purl.org/dc/terms/> .
@prefix gndo: <http://d-nb.info/standards/elementset/gnd#> .
@prefix lv:   <http://purl.org/lobid/lv#> .
@prefix mo: <http://purl.org/ontology/mo/> .
@prefix owl:   <http://www.w3.org/2002/07/owl#> .
@prefix rdau: <http://rdaregistry.info/Elements/u/> .
@prefix rdf:   <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix rdfs:  <http://www.w3.org/2000/01/rdf-schema#> .
@prefix schema: <http://schema.org/> .
@prefix skos: <http://www.w3.org/2004/02/skos/core#> .
@prefix wdrs: <http://www.w3.org/2007/05/powder-s#> .
@prefix xsd:   <http://www.w3.org/2001/XMLSchema#> .
```

**Comment by fsteeg** [comment]
Redeployed with custom prefixes:

http://stage.lobid.org/resources/HT018472857.ttl

And serving JSON-LD as application/json:

http://stage.lobid.org/resources/HT018472857.json

**Comment by ChristophEwertowski** [comment]
+1 For Adrians comment and Fabians changes

**Comment by acka47** [comment]
+1

**Comment by fsteeg** [comment]
Reopening for corresponding implementation in lobid-organisations.
